docs/languages/csharp/expression-trees-interpreting.md

```diff
@@ -0,0 +1,579 @@
+        // Visit each parameter:
+        foreach (var argumentExpression in node.Parameters)
+        {
+            var argumentVisitor = Visitor.CreateFromExpression(argumentExpression);
+            argumentVisitor.Visit(prefix + "\t");
+        }
+        Console.WriteLine($"{prefix}The expression body is:");
+        // Visit the body:
+        var bodyVisitor = Visitor.CreateFromExpression(node.Body);
+        bodyVisitor.Visit(prefix + "\t");
+    }
+}
+
+// Binary Expression Visitor:
+public class BinaryVisitor : Visitor
+{
+    private readonly BinaryExpression node;
+    public BinaryVisitor(BinaryExpression node) : base(node)
+    {
+        this.node = node;
+    }
+
+    public override void Visit(string prefix)
+    {
+        Console.WriteLine($"{prefix}This binary expression is a {NodeType} expression");
+        var left = Visitor.CreateFromExpression(node.Left);
+        Console.WriteLine($"{prefix}The Left argument is:");
+        left.Visit(prefix + "\t");
+        var right = Visitor.CreateFromExpression(node.Right);
+        Console.WriteLine($"{prefix}The Right argument is:");
+        right.Visit(prefix + "\t");
+    }
+}
+
+// Parameter visitor:
+public class ParameterVisitor : Visitor
+{
+    private readonly ParameterExpression node;
+    public ParameterVisitor(ParameterExpression node) : base(node)
+    {
+        this.node = node;
+    }
+
+    public override void Visit(string prefix)
+    {
+        Console.WriteLine($"{prefix}This is an {NodeType} expression type");
+        Console.WriteLine($"{prefix}Type: {node.Type.ToString()}, Name: {node.Name}, ByRef: {node.IsByRef}");
+    }
+}
+```
+
+This algorithm is the basis of an algorithm that can visit
+any arbitrary `LambdaExpression`. There are a lot of holes,
+namely that the code I created only looks for a very small
+sample of the possible sets of expression tree nodes that
+it may encounter. However, you can still learn quite a bit
+from what it produces. (The default case in the `Visitor.CreateFromExpression`
+method prints a message to the error console when a new node type
+is encountered. That way, you know to add a new expression type.)
+
+When you run this visitor on the addition expression shown above, you get the
+following output:
+
+```
+This expression is a Lambda expression type
+The name of the lambda is <null>
+The return type is System.Int32
+The expression has 2 arguments. They are:
+        This is an Parameter expression type
+        Type: System.Int32, Name: a, ByRef: False
+        This is an Parameter expression type
+        Type: System.Int32, Name: b, ByRef: False
+The expression body is:
+        This binary expression is a Add expression
+        The Left argument is:
+                This is an Parameter expression type
+                Type: System.Int32, Name: a, ByRef: False
+        The Right argument is:
+                This is an Parameter expression type
+                Type: System.Int32, Name: b, ByRef: False
+```
+
```

Comment:
I'd put a blurb here about how you can visit many more different types of expressions now that you've got a more general visitor implementation.